AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -469,7 +471,22 @@ - (void)performCosmosDbOperationWithPartition:(NSString *)partition
                                                                               httpMethod:httpMethod
                                                                                     body:body
                                                                        additionalHeaders:additionalHeaders
-                                                                       completionHandler:completionHandler];
+                                                                       completionHandler:^(NSData *_Nullable data,
+                                                                                           NSHTTPURLResponse *_Nullable response,
+                                                                                           NSError *_Nullable cosmosDbError) {
+                                                                         if (!cosmosDbError &&
+                                                                             ![MSHttpUtil isSuccessStatusCode:response.statusCode]) {
```

Comment:
I see only 1 new test but for 4 code branches here.

Reply:
All other branches are covered by other tests already.
This is handling edge case and the unit test tests the edge case.

Reply:
ok